### PR TITLE
Add a subs banner test that gets tracked in Optimize

### PR DESCRIPTION
**Pull request by jfsoul**
## Why are you doing this?
We want to run a test on dotcom with alternative copy on the subs banner and have that tracked by Optimize.  As we don't seem to have used the `independent` flag for a long time (and I don't trust its reliability) I've added a way for the a/b test inclusion to be picked out from the acquisition data. N.B. other epic tests don't need to do this because the a/b test data is passed through to the lake and tracked there, but we need it earlier for Optimize.

[**Trello Card**](https://trello.com/c/WrDoBRvH/2815-dummy-a-b-test-for-the-subs-banner-in-prep-for-the-6th-jan-test)

### CODE REVIEW

File: support-frontend/assets/helpers/abTests/abtest.js [hunk 1]
```diff
@@ -284,7 +284,7 @@ function getParticipations(
     if (testId in currentParticipation) {
       participations[testId] = currentParticipation[testId];
     } else if (userInTest(test, testId, mvtId, country, countryGroupId, acquisitionDataTest)) {
-      const variantIndex = assignUserToVariant(mvtId, test);
+      const variantIndex = assignUserToVariant(mvtId, test, acquisitionDataTest);
```
[the remark this answers is not in the document]
Comment (lucymonie) [reply support-frontend/assets/helpers/abTests/abtest.js]: 👍

File: support-frontend/assets/helpers/abTests/abtest.js [hunk 2]
```diff
@@ -51,6 +52,11 @@ type Audiences = {
   [IsoCountry | CountryGroupId | 'ALL']: Audience
 };
 
+type AcquisitionABTest = {
```
Comment (rupertbates) [inline support-frontend/assets/helpers/abTests/abtest.js]: This type already exists in `acquisitions.js`

File: support-frontend/assets/helpers/abTests/abtest.js [hunk 3]
```diff
@@ -192,10 +231,22 @@ const trackOptimizeExperiment = (optimizeId: string, variants: Variant[], varian
...
 ): number {
-  const { independent, seed } = test;
+  const { referrerControlled, seed } = test;
 
-  return randomNumber(mvtId, independent, seed) % test.variants.length;
+  if (referrerControlled && acquisitionDataTest != null) {
+    const acquisitionVariant = acquisitionDataTest.variant;
+    const index = test.variants.findIndex(variant => variant.id === acquisitionVariant);
+    if (!index) {
+      console.error('Variant not found for A/B test in acquistion data');
```
Comment (rupertbates) [inline support-frontend/assets/helpers/abTests/abtest.js]: typo in 'acquisition'

File: support-frontend/assets/helpers/abTests/abtest.js [hunk 4]
```diff
@@ -192,10 +231,22 @@ const trackOptimizeExperiment = (optimizeId: string, variants: Variant[], varian
...
+    }
+    return index;
+  } else if (referrerControlled && acquisitionDataTest === null) {
+    console.error('A/B test expects acquistion data but none was provided');
```
Comment (rupertbates) [inline support-frontend/assets/helpers/abTests/abtest.js]: Same typo

File: support-frontend/assets/helpers/abTests/abtest.js (hunk 3 above)
Comment (lucymonie) [reply support-frontend/assets/helpers/abTests/abtest.js]: There's the need for that keyboard shortcut! 😄

File: support-frontend/assets/helpers/abTests/abtest.js (hunk 2 above)
Comment (lucymonie) [reply support-frontend/assets/helpers/abTests/abtest.js]: Thanks Rupert! 👍

File: support-frontend/assets/helpers/abTests/abtest.js (hunk 4 above)
Comment (lucymonie) [reply support-frontend/assets/helpers/abTests/abtest.js]: 😢

File: support-frontend/package.json [hunk 5]
```diff
@@ -55,7 +55,8 @@
       "ophan(.*)": "<rootDir>/node_modules/ophan-tracker-js/build/ophan.support"
     },
     "verbose": true,
-    "testEnvironment": "./jestEnvironment"
+    "testEnvironment": "./jestEnvironment",
+    "testURL": "https://www.somthing.com/test.html"
```
Comment (damien-shaw-guardian) [inline support-frontend/package.json]: What is this used for? @lucymonie

**Pull request merged**